components/support/utils/src/main/java/mozilla/components/support/utils/intents.kt

```diff
@@ -0,0 +1,25 @@
+@file:JvmName("IntentUtils")
+
+package mozilla.components.support.utils
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+
+/**
+ * Create a [PendingIntent] instance to run a certain service described with the [Intent].
+ *
+ * This method will allow you to launch a service that will be able to overpass
+ * [background service limitations](https://developer.android.com/about/versions/oreo/background#services)
+ * introduced in Android Oreo.
+ *
+ * @param context an [Intent] to start a service.
+ */
+@JvmName("createPendingIntentForLaunchService")
+fun Intent.asPendingIntentForLaunchService(context: Context): PendingIntent =
```

Comment:
nit: I think the name should mention that it is a **foreground** service!

Reply:
Naming ideas are welcomed. I don't think that we can use "foreground" here because it's decieving for older android versions. Technically this is just a **service-that-can-start-activities-from-background**.

Reply:
I think using foreground makes still sense here. It will gracefully downgrade on services before Q - similar to `ContextCompat.startForegroundService()`.

Reply:
This is a tough one.  Both makes sense.  I would lean towards adding "foreground" because most users of this API will be supporting O+ devices and would need to be aware that this will start a foreground service.

Reply:
Deal. I'll change the name, thanks. :)

Reply:
:heavy_check_mark: Done.